Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -17,22 +17,39 @@ sealed class SessionPool : IDisposable
     {
         private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<SessionPool>();
         private static readonly object s_sessionPoolLock = new object();
-        private readonly List<SFSession> _sessions;
+
+        private readonly List<SFSession> _idleSessions;
         private int _maxPoolSize;
         private long _timeout;
         private const int MaxPoolSize = 10;
         private const long Timeout = 3600;
+        internal string ConnectionString;
+        internal SecureString Password;
         private bool _pooling = true;
+        private bool _allowExceedMaxPoolSize = true;
+        internal static ISessionFactory SessionFactory = new SessionFactory();
```

Comment:
Good idea, will do

Reply:
On second thought, it would be better but requires passing the same factory through multiple layers of code. Is it really worth? From test perspective it will be just injected into another class using some static method. Same as now.